pkg/client/typed/scaler/scaler_test.go

```diff
@@ -0,0 +1,247 @@
+			scaleRef: extensions.SubresourceReference{
+				Kind:        "ReplicationController",
+				Name:        "extrc",
+				APIVersion:  "extensions/v1beta1",
+				Subresource: "scale",
+			},
+			resp:     getJSON("extensions/v1beta1", "Scale", "extrc", replicas, `{"foo": "bar"}`),
+			replicas: replicas,
+			selector: `foo=bar`,
+		},
+		{
+			namespace: "testns",
+			name:      "extdep",
+			path:      "/apis/extensions/v1beta1/namespaces/testns/deployments/extdep/scale",
+			scaleRef: extensions.SubresourceReference{
+				Kind:        "Deployment",
+				Name:        "extdep",
+				APIVersion:  "extensions/v1beta1",
+				Subresource: "scale",
+			},
+			resp:     getJSON("autoscaling/v1", "Scale", "extdep", replicas, fmt.Sprintf("%q", labelSelector.String())),
+			replicas: replicas,
+			selector: labelSelector.String(),
+		},
+	}
+	for _, tc := range tcs {
+		gv := &unversioned.GroupVersion{Group: "extensions", Version: "v1beta1"}
+		cl, srv, err := getClientServer(gv, func(w http.ResponseWriter, r *http.Request) {
+			if r.Method != "GET" {
+				t.Errorf("Get(%q) got HTTP method %s. wanted GET", tc.name, r.Method)
+			}
+
+			if r.URL.Path != tc.path {
+				t.Errorf("Get(%q) got path %s. wanted %s", tc.name, r.URL.Path, tc.path)
+			}
+
+			w.Write(tc.resp)
+		})
+		if err != nil {
+			t.Errorf("unexpected error when creating client: %v", err)
+			continue
+		}
+		defer srv.Close()
+
+		got, err := cl.Get(tc.scaleRef, tc.namespace)
+		if err != nil {
+			t.Errorf("unexpected error when getting %q: %v", tc.name, err)
+			continue
+		}
+
+		if r, err := got.Replicas(); err != nil {
+			t.Errorf("Get(%q) unexpected replicas error: %v", tc.name, err)
+		} else if r != tc.replicas {
+			t.Errorf("Get(%q) want replicas: %d\tgot: %d", tc.name, tc.replicas, r)
+		}
+		if s, err := got.Selector(); err != nil {
+			t.Errorf("Get(%q) unexpected selector error: %v", tc.name, err)
+		} else if s.String() != tc.selector {
+			t.Errorf("Get(%q) want selector: %s\tgot: %s", tc.name, tc.selector, s)
+		}
+	}
+}
+
+func TestSetReplicas(t *testing.T) {
+	initReplicas := 5
+	finalReplicas := 10
+	sel := &unversioned.LabelSelector{
+		MatchLabels: map[string]string{"foo": "bar"},
+		MatchExpressions: []unversioned.LabelSelectorRequirement{
+			{
+				Key:      "lkfoo",
+				Operator: unversioned.LabelSelectorOpIn,
+				Values:   []string{"lbar", "lbaz"},
+			},
+			{
+				Key:      "lkexist",
+				Operator: unversioned.LabelSelectorOpExists,
+			},
+		},
+	}
+	labelSelector, err := unversioned.LabelSelectorAsSelector(sel)
+	if err != nil {
+		t.Errorf("failed to convert label selector to selector: %v", err)
+	}
+
+	tcs := []struct {
+		namespace string
+		name      string
+		path      string
+		scaleRef  extensions.SubresourceReference
+		resp      []byte
+		replicas  int
+		selector  string
+	}{
+		{
+			namespace: "testns",
+			name:      "extrc",
+			path:      "/apis/extensions/v1beta1/namespaces/testns/replicationcontrollers/extrc/scale",
```

Comment:
I thought we started mounting this in `/api/v1/namespaces/testns/replicationcontrollers/extrc/scale`, too? Can you test that location also, if so?